jobs/src/main/scala/dcos/metronome/jobspec/impl/JobSpecDependencyActor.scala

```diff
@@ -78,10 +81,13 @@ class JobSpecDependencyActor(initSpec: JobSpec, runService: JobRunService) exten
         log.info(
           s"Start next run of job ${spec.id}, all parents finished successfully lastParentRuns=${lastParentRuns.values}"
         )
-        runService.startJobRun(initSpec).pipeTo(self)
+        runService.startJobRun(currentSpec).pipeTo(self)
         context.become(starting)
       }
 
+    case ev: Event.JobSpecUpdated =>
```

Comment:
Any idea what happens when the job is deleted? Does this come across as an updated event?

Should we terminate the actor in that case?

Reply:
it was my take that this is handled [here](https://github.com/dcos/metronome/blob/b7b4a2c7569592a22fa4d7e1f06a92330df4d30a/jobs/src/main/scala/dcos/metronome/jobspec/impl/JobSpecServiceActor.scala#L167-L179).